components/Typography/src/MDCFontScaler.m

```diff
@@ -0,0 +1,337 @@
+        UIContentSizeCategoryAccessibilityMedium : @22,
+        UIContentSizeCategoryAccessibilityLarge : @24,
+        UIContentSizeCategoryAccessibilityExtraLarge : @26,
+        UIContentSizeCategoryAccessibilityExtraExtraLarge : @28,
+        UIContentSizeCategoryAccessibilityExtraExtraExtraLarge : @30
+      };
+      _scalingCurve = scalingCurve;
+    } else if ([MDCTextStyleCaption isEqualToString:textStyle]) {
+      NSDictionary<UIContentSizeCategory, NSNumber *> *scalingCurve = @{
+        UIContentSizeCategoryExtraSmall : @11,
+        UIContentSizeCategorySmall : @11,
+        UIContentSizeCategoryMedium : @11,
+        UIContentSizeCategoryLarge : @12,
+        UIContentSizeCategoryExtraLarge : @14,
+        UIContentSizeCategoryExtraExtraLarge : @16,
+        UIContentSizeCategoryExtraExtraExtraLarge : @18,
+        UIContentSizeCategoryAccessibilityMedium : @20,
+        UIContentSizeCategoryAccessibilityLarge : @22,
+        UIContentSizeCategoryAccessibilityExtraLarge : @24,
+        UIContentSizeCategoryAccessibilityExtraExtraLarge : @26,
+        UIContentSizeCategoryAccessibilityExtraExtraExtraLarge : @28
+      };
+      _scalingCurve = scalingCurve;
+    } else if ([MDCTextStyleOverline isEqualToString:textStyle]) {
+      NSDictionary<UIContentSizeCategory, NSNumber *> *scalingCurve = @{
+        UIContentSizeCategoryExtraSmall : @8,
+        UIContentSizeCategorySmall : @8,
+        UIContentSizeCategoryMedium : @9,
+        UIContentSizeCategoryLarge : @10,
+        UIContentSizeCategoryExtraLarge : @12,
+        UIContentSizeCategoryExtraExtraLarge : @14,
+        UIContentSizeCategoryExtraExtraExtraLarge : @16,
+        UIContentSizeCategoryAccessibilityMedium : @18,
+        UIContentSizeCategoryAccessibilityLarge : @20,
+        UIContentSizeCategoryAccessibilityExtraLarge : @22,
+        UIContentSizeCategoryAccessibilityExtraExtraLarge : @24,
+        UIContentSizeCategoryAccessibilityExtraExtraExtraLarge : @26
+      };
+      _scalingCurve = scalingCurve;
+    } else {
+      // If nothing matches, return the metrics for MDCTextStyleBody1
+      NSDictionary<UIContentSizeCategory, NSNumber *> *scalingCurve = @{
+        UIContentSizeCategoryExtraSmall : @13,
+        UIContentSizeCategorySmall : @14,
+        UIContentSizeCategoryMedium : @15,
+        UIContentSizeCategoryLarge : @16,
+        UIContentSizeCategoryExtraLarge : @18,
+        UIContentSizeCategoryExtraExtraLarge : @20,
+        UIContentSizeCategoryExtraExtraExtraLarge : @22,
+        UIContentSizeCategoryAccessibilityMedium : @26,
+        UIContentSizeCategoryAccessibilityLarge : @30,
+        UIContentSizeCategoryAccessibilityExtraLarge : @34,
+        UIContentSizeCategoryAccessibilityExtraExtraLarge : @38,
+        UIContentSizeCategoryAccessibilityExtraExtraExtraLarge : @42
+      };
+      _scalingCurve = scalingCurve;
+    }
+  }
+
+  return self;
+}
+
+- (UIFont *)scalableFontWithFont:(UIFont *)font {
```

Comment:
Looks like this should be returning a **scalable** font, not a **scaled** font.  Please do not scale the returned font, but only attached the relevant scaling curves.

Reply:
This behavior is intentional to match UIKit's UIFontMetrics.
This is documented in the header.

Reply:
Please either rename this to `scaledFontForFont:` to match UIFontMetrics or have it return a **scalable but unscaled** font.